tensor2tensor/data_generators/wmt.py

```diff
@@ -69,14 +52,13 @@ def is_character_level(self):
   def targeted_vocab_size(self):
     raise NotImplementedError()  # Not needed if self.is_character_level.
 
-  @property
-  def train_generator(self):
-    """Generator; takes tmp_dir, is_training, possibly targeted_vocab_size."""
+  def train_generator(self, tmp_dir, is_training):
```

Comment:
Unluckily these arguments now include data_dir, this will need updating. Also, they are different for character models, but it could be just an empty arg (_). Thanks for making this nicer!